main/modes/mode_gamepad.c

```diff
@@ -969,20 +969,23 @@ void gamepadReportStateToHost(void)
 
 static bool saveGamepadToggleSettings(gamepadToggleSettings_t* toggleSettings)
 {
-    return writeNvsBlob(KEY_GAMEPAD_TOGGLES, toggleSettings, sizeof(gamepadToggleSettings_t));
+    int32_t val = ((toggleSettings->accelOn & 1) << 1) | ((toggleSettings->touchAnalogOn & 1) << 0);
```

Comment:
What's the advantage to this change? Doing this bitpacked math (and undoing it on the other end) seems more complicated and harder to maintain than reading and writing the struct driectly.

If you're _that_ concerned about space, you should bitpack your struct instead: https://stackoverflow.com/questions/25822679/packed-bit-fields-in-c-structures-gcc

Reply:
Storing anything as a blob uses a minimum of 96 bytes of NVS, whereas storing as int32 uses a minimum of 32 bytes of NVS. Would bitpacking my struct still allow me to stick it into an int32 and expand it without losing existing data?

Reply:
I see the 32 bytes for a 32 bit integer here: https://docs.espressif.com/projects/esp-idf/en/v4.4.3/esp32s2/api-reference/storage/nvs_flash.html#structure-of-entry. I don't see where 96 bytes is coming from, but I do see that blobs have a bit more overhead. This may be a job for a union. You can have a union with the `struct` and an `int32_t`, and as long as the struct is 32 bits or fewer, you can read/write the `int32_t` to NVS and access the good stuff via the `struct`. It's not the cleanest, but anything is better than mathing individual fields.

Reply:
Honestly, sometimes a little overhead is worth it for maintainability and cleanliness. We don't have to go overboard with optimizations.

Reply:
Unless you want to rewrite everything in assembly!

Reply:
Every blob in NVS uses a 32-byte index entry to index its chunks, a 32-byte chunk-data entry per chunk, and however many 32-byte data-only entries the actual data requires. Think of chunks as NTFS fragments, at least that's the way they're presented on the documentation page.

Reply:
I'm trying to see if we can safely fit a 5th paint slot, so I want to keep my unnecessary NVS footprint to a minimum.

Reply:
Switched from bitpacking to a union, as suggested.